diff --git a/node_reference/node_reference.module b/node_reference/node_reference.module
index 95560ed..2d604c9 100644
--- a/node_reference/node_reference.module
+++ b/node_reference/node_reference.module
@@ -741,14 +741,18 @@ function _node_reference_options($field, $flat = TRUE) {
 
   $options = array();
   foreach ($references as $key => $value) {
+    // The label, displayed in selects and checkboxes/radios, should have HTML
+    // entities unencoded. The widgets (core's options.module) take care of
+    // applying the relevant filters (strip_tags() or filter_xss()).
+    $label = html_entity_decode($value['rendered'], ENT_QUOTES);
     if (empty($value['group']) || $flat) {
-      $options[$key] = $value['rendered'];
+      $options[$key] = $label;
     }
     else {
       // The group name, displayed in selects, cannot contain tags, and should
       // have HTML entities unencoded.
       $group = html_entity_decode(strip_tags($value['group']), ENT_QUOTES);
-      $options[$group][$key] = $value['rendered'];
+      $options[$group][$key] = $label;
     }
   }
 
diff --git a/user_reference/user_reference.module b/user_reference/user_reference.module
index c8b26b5..c480b71 100644
--- a/user_reference/user_reference.module
+++ b/user_reference/user_reference.module
@@ -766,14 +766,18 @@ function _user_reference_options($field, $flat = TRUE) {
 
   $options = array();
   foreach ($references as $key => $value) {
+    // The label, displayed in selects and checkboxes/radios, should have HTML
+    // entities unencoded. The widgets (core's options.module) take care of
+    // applying the relevant filters (strip_tags() or filter_xss()).
+    $label = html_entity_decode($value['rendered'], ENT_QUOTES);
     if (empty($value['group']) || $flat) {
-      $options[$key] = $value['rendered'];
+      $options[$key] = $label;
     }
     else {
       // The group name, displayed in selects, cannot contain tags, and should
       // have HTML entities unencoded.
       $group = html_entity_decode(strip_tags($value['group']), ENT_QUOTES);
-      $options[$group][$key] = $value['rendered'];
+      $options[$group][$key] = $label;
     }
   }
 
